### PR TITLE
Fixed resource_aci_l3extinstp idempotency and relationship attribute import issues

### PR DESCRIPTION
**Pull request by sajagana**
Note:

Fixed External EPGs resource issues, updated existing documents and example files.

### CODE REVIEW

File: website/docs/d/external_network_instance_profile.html.markdown [hunk 1]
```diff
@@ -11,28 +11,46 @@ description: |-
...
+* `id` - Attribute id set to the Dn of the External Network Instance Profile.
+* `annotation` - (Optional) Annotation of the External Network Instance Profile object.
+* `exception_tag` - (Optional) Exception tag of the External Network Instance Profile object.
+* `flood_on_encap` - (Optional) Control at EPG level if the traffic L2 Multicast/Broadcast and Link Local Layer should be flooded only on ENCAP or based on bridg-domain settings.
+* `match_t` - (Optional) The provider label match criteria of the External Network Instance Profile object.
+* `name_alias` - (Optional) Name alias of the External Network Instance Profile object.
+* `pref_gr_memb` - (Optional) Represents parameter used to determine if EPg is part of a group that does not a contract for communication.
+* `prio` - (Optional) The QoS priority class identifier of the External Network Instance Profile object.
+* `target_dscp` - (Optional) The target differentiated services code point (DSCP) of the path attached to the layer 3 outside profile.
+* `relation_fv_rs_sec_inherited` - (Optional) Relation to class fvEPg. Cardinality - N_TO_M. Type - Set of String.
```
Comment (lhercot) [inline website/docs/d/external_network_instance_profile.html.markdown:48]: Relation to EPGs to be used as Contract Masters (class fvEPg).

File: website/docs/d/external_network_instance_profile.html.markdown [hunk 2]
```diff
@@ -11,28 +11,46 @@ description: |-
...
+* `relation_fv_rs_prov` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
```
Comment (lhercot) [inline website/docs/d/external_network_instance_profile.html.markdown:49]: Relation to Provided Contracts (class vzBrCP).

File: website/docs/d/external_network_instance_profile.html.markdown [hunk 3]
```diff
@@ -11,28 +11,46 @@ description: |-
...
+* `relation_fv_rs_cons_if` - (Optional) Relation to class vzCPIf. Cardinality - N_TO_M. Type - Set of String.
+* `relation_l3ext_rs_inst_p_to_profile` - (Optional) Relation to class rtctrlProfile. Cardinality - N_TO_M. Type: Block.
+  * tn_rtctrl_profile_name - (Deprecated) Name of the Route map for import and export route control.
+  * tn_rtctrl_profile_dn - (Optional) Distinguished name of the Route map for import and export route control.
+  * direction - (Optional) Direction of the Route Control Profile.
+* `relation_fv_rs_cons` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
```
Comment (lhercot) [inline website/docs/d/external_network_instance_profile.html.markdown:55]: Relation to Consumed Contracts (class vzBrCP).

File: website/docs/d/external_network_instance_profile.html.markdown [hunk 4]
```diff
@@ -11,28 +11,46 @@ description: |-
...
+* `exception_tag` - (Optional) Exception tag of the External Network Instance Profile object.
+* `flood_on_encap` - (Optional) Control at EPG level if the traffic L2 Multicast/Broadcast and Link Local Layer should be flooded only on ENCAP or based on bridg-domain settings.
+* `match_t` - (Optional) The provider label match criteria of the External Network Instance Profile object.
+* `name_alias` - (Optional) Name alias of the External Network Instance Profile object.
+* `pref_gr_memb` - (Optional) Represents parameter used to determine if EPg is part of a group that does not a contract for communication.
+* `prio` - (Optional) The QoS priority class identifier of the External Network Instance Profile object.
+* `target_dscp` - (Optional) The target differentiated services code point (DSCP) of the path attached to the layer 3 outside profile.
+* `relation_fv_rs_sec_inherited` - (Optional) Relation to class fvEPg. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_prov` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_cons_if` - (Optional) Relation to class vzCPIf. Cardinality - N_TO_M. Type - Set of String.
```
Comment (lhercot) [inline website/docs/d/external_network_instance_profile.html.markdown:50]: Relation to Provided Contract Interfaces (class vzCPIf).

File: website/docs/d/external_network_instance_profile.html.markdown [hunk 5]
```diff
@@ -11,28 +11,46 @@ description: |-
...
+* `relation_l3ext_rs_inst_p_to_profile` - (Optional) Relation to class rtctrlProfile. Cardinality - N_TO_M. Type: Block.
```
Comment (lhercot) [inline website/docs/d/external_network_instance_profile.html.markdown:51]: Relation to Route Control Profiles (class rtctrlProfile)

File: website/docs/d/external_network_instance_profile.html.markdown [hunk 6]
```diff
@@ -11,28 +11,46 @@ description: |-
...
+  * tn_rtctrl_profile_name - (Deprecated) Name of the Route map for import and export route control.
+  * tn_rtctrl_profile_dn - (Optional) Distinguished name of the Route map for import and export route control.
+  * direction - (Optional) Direction of the Route Control Profile.
+* `relation_fv_rs_cons` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_prot_by` - (Optional) Relation to class vzTaboo. Cardinality - N_TO_M. Type - Set of String.
```
Comment (lhercot) [inline website/docs/d/external_network_instance_profile.html.markdown:56]: Relation to Taboo Contracts (vzTaboo).

File: website/docs/r/external_network_instance_profile.html.markdown [hunk 7]
```diff
@@ -11,44 +11,68 @@ description: |-
...
+* `relation_fv_rs_sec_inherited` - (Optional) Relation to class fvEPg. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_prov` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_cons_if` - (Optional) Relation to class vzCPIf. Cardinality - N_TO_M. Type - Set of String.
+* `relation_l3ext_rs_inst_p_to_profile` - (Optional) Relation to class rtctrlProfile. Every direction(export/import) allows only one object. Cardinality - N_TO_M. Type: Block.
+  * tn_rtctrl_profile_name - (Deprecated) Name of the Route map for import and export route control.
+  * tn_rtctrl_profile_dn - (Optional) Distinguished name of the Route map for import and export route control.
+  * direction - (Optional) Direction of the Route Control Profile. Allowed values are "export" and "import".
+* `relation_fv_rs_cons` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_prot_by` - (Optional) Relation to class vzTaboo. Cardinality - N_TO_M. Type - Set of String.
```
Comment (lhercot) [inline website/docs/r/external_network_instance_profile.html.markdown:75]: Relation to Taboo Contracts (vzTaboo).

File: website/docs/r/external_network_instance_profile.html.markdown [hunk 8]
```diff
@@ -11,44 +11,68 @@ description: |-
...
+* `relation_fv_rs_sec_inherited` - (Optional) Relation to class fvEPg. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_prov` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_cons_if` - (Optional) Relation to class vzCPIf. Cardinality - N_TO_M. Type - Set of String.
+* `relation_l3ext_rs_inst_p_to_profile` - (Optional) Relation to class rtctrlProfile. Every direction(export/import) allows only one object. Cardinality - N_TO_M. Type: Block.
+  * tn_rtctrl_profile_name - (Deprecated) Name of the Route map for import and export route control.
+  * tn_rtctrl_profile_dn - (Optional) Distinguished name of the Route map for import and export route control.
+  * direction - (Optional) Direction of the Route Control Profile. Allowed values are "export" and "import".
+* `relation_fv_rs_cons` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
```
Comment (lhercot) [inline website/docs/r/external_network_instance_profile.html.markdown:74]: Relation to Consumed Contracts (class vzBrCP).

File: website/docs/r/external_network_instance_profile.html.markdown [hunk 9]
```diff
@@ -11,44 +11,68 @@ description: |-
...
+* `prio` - (Optional) The QoS priority class identifier of the External Network Instance Profile object. Allowed values are "unspecified", "level1", "level2", "level3", "level4", "level5" and "level6". Default is "unspecified".
+* `target_dscp` - (Optional) The target differentiated services code point (DSCP) of the path attached to the layer 3 outside profile. Allowed values are "CS0", "CS1", "AF11", "AF12", "AF13", "CS2", "AF21", "AF22", "AF23", "CS3", "AF31", "AF32", "AF33", "CS4", "AF41", "AF42", "AF43", "CS5", "VA", "EF", "CS6", "CS7" and "unspecified". Default is "unspecified".
+* `relation_fv_rs_sec_inherited` - (Optional) Relation to class fvEPg. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_prov` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_cons_if` - (Optional) Relation to class vzCPIf. Cardinality - N_TO_M. Type - Set of String.
+* `relation_l3ext_rs_inst_p_to_profile` - (Optional) Relation to class rtctrlProfile. Every direction(export/import) allows only one object. Cardinality - N_TO_M. Type: Block.
```
Comment (lhercot) [inline website/docs/r/external_network_instance_profile.html.markdown:70]: Relation to Route Control Profiles (class rtctrlProfile)

File: website/docs/r/external_network_instance_profile.html.markdown [hunk 10]
```diff
@@ -11,44 +11,68 @@ description: |-
...
+* `pref_gr_memb` - (Optional) Represents parameter used to determine if EPg is part of a group that does not a contract for communication. Allowed values are "include" and "exclude". Default is "exclude".
+* `prio` - (Optional) The QoS priority class identifier of the External Network Instance Profile object. Allowed values are "unspecified", "level1", "level2", "level3", "level4", "level5" and "level6". Default is "unspecified".
+* `target_dscp` - (Optional) The target differentiated services code point (DSCP) of the path attached to the layer 3 outside profile. Allowed values are "CS0", "CS1", "AF11", "AF12", "AF13", "CS2", "AF21", "AF22", "AF23", "CS3", "AF31", "AF32", "AF33", "CS4", "AF41", "AF42", "AF43", "CS5", "VA", "EF", "CS6", "CS7" and "unspecified". Default is "unspecified".
+* `relation_fv_rs_sec_inherited` - (Optional) Relation to class fvEPg. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_prov` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_cons_if` - (Optional) Relation to class vzCPIf. Cardinality - N_TO_M. Type - Set of String.
```
Comment (lhercot) [inline website/docs/r/external_network_instance_profile.html.markdown:69]: Relation to Provided Contract Interfaces (class vzCPIf).

File: website/docs/r/external_network_instance_profile.html.markdown [hunk 11]
```diff
@@ -11,44 +11,68 @@ description: |-
...
+* `name_alias` - (Optional) Name alias of the External Network Instance Profile object.
+* `pref_gr_memb` - (Optional) Represents parameter used to determine if EPg is part of a group that does not a contract for communication. Allowed values are "include" and "exclude". Default is "exclude".
+* `prio` - (Optional) The QoS priority class identifier of the External Network Instance Profile object. Allowed values are "unspecified", "level1", "level2", "level3", "level4", "level5" and "level6". Default is "unspecified".
+* `target_dscp` - (Optional) The target differentiated services code point (DSCP) of the path attached to the layer 3 outside profile. Allowed values are "CS0", "CS1", "AF11", "AF12", "AF13", "CS2", "AF21", "AF22", "AF23", "CS3", "AF31", "AF32", "AF33", "CS4", "AF41", "AF42", "AF43", "CS5", "VA", "EF", "CS6", "CS7" and "unspecified". Default is "unspecified".
+* `relation_fv_rs_sec_inherited` - (Optional) Relation to class fvEPg. Cardinality - N_TO_M. Type - Set of String.
+* `relation_fv_rs_prov` - (Optional) Relation to class vzBrCP. Cardinality - N_TO_M. Type - Set of String.
```
Comment (lhercot) [inline website/docs/r/external_network_instance_profile.html.markdown:68]: Relation to Provided Contracts (class vzBrCP).

File: website/docs/r/external_network_instance_profile.html.markdown [hunk 12]
```diff
@@ -11,44 +11,68 @@ description: |-
...
+* `name_alias` - (Optional) Name alias of the External Network Instance Profile object.
+* `pref_gr_memb` - (Optional) Represents parameter used to determine if EPg is part of a group that does not a contract for communication. Allowed values are "include" and "exclude". Default is "exclude".
+* `prio` - (Optional) The QoS priority class identifier of the External Network Instance Profile object. Allowed values are "unspecified", "level1", "level2", "level3", "level4", "level5" and "level6". Default is "unspecified".
+* `target_dscp` - (Optional) The target differentiated services code point (DSCP) of the path attached to the layer 3 outside profile. Allowed values are "CS0", "CS1", "AF11", "AF12", "AF13", "CS2", "AF21", "AF22", "AF23", "CS3", "AF31", "AF32", "AF33", "CS4", "AF41", "AF42", "AF43", "CS5", "VA", "EF", "CS6", "CS7" and "unspecified". Default is "unspecified".
+* `relation_fv_rs_sec_inherited` - (Optional) Relation to class fvEPg. Cardinality - N_TO_M. Type - Set of String.
```
Comment (lhercot) [inline website/docs/r/external_network_instance_profile.html.markdown:67]: Relation to EPGs to be used as Contract Masters (class fvEPg).

File: website/docs/r/external_network_instance_profile.html.markdown [hunk 13]
```diff
@@ -11,44 +11,68 @@ description: |-
...
+* `l3_outside_dn` - (Required) Distinguished name of the parent L3Outside object.
+* `name` - (Required) Name of the External Network Instance Profile object.
+* `annotation` - (Optional) Annotation of the External Network Instance Profile object.
+* `exception_tag` - (Optional) Exception tag of the External Network Instance Profile object. Allowed value range is "0" - "512".
+* `flood_on_encap` - (Optional) Control at EPG level if the traffic L2 Multicast/Broadcast and Link Local Layer should be flooded only on ENCAP or based on bridg-domain settings. Allowed values are "disabled" and "enabled". Default value is "disabled".
+* `match_t` - (Optional) The provider label match criteria of the External Network Instance Profile object. Allowed values are "All", "AtleastOne", "AtmostOne" and "None". Default is "AtleastOne".
+* `name_alias` - (Optional) Name alias of the External Network Instance Profile object.
+* `pref_gr_memb` - (Optional) Represents parameter used to determine if EPg is part of a group that does not a contract for communication. Allowed values are "include" and "exclude". Default is "exclude".
```
Comment (lhercot) [inline website/docs/r/external_network_instance_profile.html.markdown:64]: if an External EPG is part of a group that does not require a contract for communication

**Review by lhercot — CHANGES_REQUESTED** [review]
Please also rebase on latest master

**Review by lhercot — APPROVED** [review]
LGTM

**Review by akinross — APPROVED** [review]
LGTM

**Pull request merged**